colcon_output/event_handler/log.py

```diff
@@ -78,9 +78,13 @@ def __call__(self, event):  # noqa: D102
 
         if isinstance(data, JobEnded):
             # Skip if the log path is /dev/null
+            # and/or is unable to get directory
             if get_log_path() is None:
                 return
```

Comment:
This check is redundant now that you added the check in `get_log_directory()`.
```suggestion
```

Reply:
I intentionally left that redundancy in case the `job.id` could be `None`.
But I agree it should not (it would fail in previous calls first).